frontend/packages/console-shared/src/selectors/common.ts

```diff
@@ -2,11 +2,11 @@ import * as _ from 'lodash';
 
 import { K8sResourceKind } from '@console/internal/module/k8s';
 
-export const getName = (value: K8sResourceKind) =>
+export const getName = <A extends K8sResourceKind>(value: A) =>
   _.get(value, 'metadata.name') as K8sResourceKind['metadata']['name'];
-export const getNamespace = (value: K8sResourceKind) =>
+export const getNamespace = <A extends K8sResourceKind>(value: A) =>
   _.get(value, 'metadata.namespace') as K8sResourceKind['metadata']['namespace'];
-export const getUID = (value: K8sResourceKind) =>
+export const getUID = <A extends K8sResourceKind>(value: A) =>
```

Comment:
I had to also change signature of these selectors to be compatible with the lookup generics